hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientInvocationServiceSupport.java

```diff
@@ -265,11 +278,16 @@ public ClientMessage getClientMessage() {
     }
 
     private class ResponseThread extends Thread {
-        private final BlockingQueue<ClientPacket> workQueue = new LinkedBlockingQueue<ClientPacket>();
+        private final BlockingQueue<ClientPacket> responseQueue;
 
         public ResponseThread(ThreadGroup threadGroup, String name, ClassLoader classLoader) {
             super(threadGroup, name);
             setContextClassLoader(classLoader);
+
+            IdleStrategy idleStrategy = client.getProperties().getBoolean(FAST)
+                    ? new BackoffIdleStrategy(IDLE_MAX_SPINS, IDLE_MAX_YIELDS, IDLE_MIN_PARK_NS, IDLE_MAX_PARK_NS)
+                    : new NoOpIdleStrategy();
```

Comment:
I made a change in the approach. I can't rely on the return value of the IdlePolicy to determine if I should block or not. 

So if no IdlePolicy is set, the consumer will block.

If an IdlePolicy is set, the consumer will not block and will follow let the IdlePolicy take care of idling.